### ISSUE TITLE
Feature Request: Next, Prev and Go show controls

### ISSUE THREAD

**Issue by cmdtv** [bitfocus/companion-module-behringer-x32#32]
Add support for Next, Prev and Go buttons to move up and down scene or cue list to control show.

**Comment by justingiffard** [comment]
I have opened a feature request on the XM32 Series Trello board to have OSC commands for this be added

**Comment by justingiffard** [comment]
Update: feature suggestion accepted and will be looked into 🥳 
https://trello.com/c/cjmYGB8r

**Comment by stevekmcc** [comment]
GO can be implemented with the current X32 firmware: 
https://community.musictribe.com/discussions/223527/311014/x32-midi-mapping

(I'll copy-paste the answer here, since who knows how long a MusicTribe URL will last!)
Paul_Vannatto wrote March 29 2021:
There is no midi or OSC command for the Go button directly. The way I have implemented the Go command in X32 Live Toolbox is as follows:
* Get current position using the /-show/prepos/current OSC command
* Send the appropriate OSC command with the current position as the value (eg. /-action/goscene ,i 15

**Comment by justingiffard** [comment]
@stevekmcc interesting, would need to take an extra step that they don't mention (I suppose they do allude to it) 
would need to use `/-prefs/show_control` to determine whether to use `/‐action/gocue`,  `/-action/goscene`  or `/‐action/gosnippet` 

technically I could use similar logic for next and previous. Am I right in saying it would just change the "highlighted" scene up or down? wonder how it handles when you go to a number above the number of scenes you have (eg you have 2 scenes and you press next 4 times... Im assuming it would go to 6 still and if you say "go" it will just ignore it... but then you would need to press previous 4 times to get it to be "in range" again) unless I check to see does scene 3 have data `/‐show/showfile/scene/2/hasdata` if yes set `/-show/prepos/current` to 3 else ignore

will look at this tomorrow though when I've had some sleep

**Comment by stevekmcc** [comment]
Yes, it looks like that's how they do it too. X32 Live Toolbox is open source, so see the latest version, https://sourceforge.net/projects/x32livetoolbox/files/Source%20Code/LiveToolbox-133-source.zip/download
file OscTxRx.pas lines 2943-3062

**Comment by justingiffard** [comment]
I always forget ugly Pascal/Delphi is... Wowza I struggle to read that 🙈

**Comment by justingiffard** [comment]
okay I think I have it implemented... Will text on either Thursday or Sunday when Im at church again
Its not pretty with needing to ensure 700 hasDatas are loaded but... if I want to check that each the next or previous has data...it has to be done (just realised while typing this I probably dont need to do that check for previous because the current highlighted thing SHOULD always have data, but still need to check preload 500 cues 100 scenes and 100 snippet for next... yay -_- thank goodness for loops I guess)
